console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/server/GwtDeviceManagementServiceImpl.java

```diff
@@ -296,7 +296,12 @@ public int compare(DeviceComponentConfiguration arg0, DeviceComponentConfigurati
                     if (ocd != null) {
                         GwtConfigComponent gwtConfig = new GwtConfigComponent();
                         gwtConfig.setId(config.getId());
-                        gwtConfig.setName(ocd.getName());
+                        if(config.getId().indexOf('.') == -1){
```

Comment:
Just a small suggestion, no change required :) 

Instead of using `aString.indexOf('x') == 1` you can use `aString.contains('x')` which is implemented using the `indexOf(...)` (e.g. ` return indexOf(s.toString()) > -1`). 

The benefit is that the code is slightly more readable :)

Reply:
I will keep that in mind. Thanks for the suggestion!